### PR TITLE
General fixes

### PR DESCRIPTION
**Pull request by pcarrott**
Improvements:
1. Increase default timeout
2. Create Timeout exception
3. Create NotationNotFound exception
4. Make AST traversals iterative

Main changes and additions:
1. Add support for new Coq keywords
2. CoqFile keeps a list of steps instead of plain text and AST
3. Each step stores the associated diagnostics
4. Handle where-notations with Unicode characters
5. Constrain regex expression for notation wildcards

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,8 +25,6 @@ from coqlspclient.coq_structs import TermType
 
 # Open Coq file
 with CoqFile(os.path.join(os.getcwd(), "examples/example.v")) as coq_file:
-    # Print AST
-    print(coq_file.ast)
```
Comment (Nfsaavedra) [inline README.md:29]: Why was this removed?

File: pylspclient/lsp_structs.py [hunk 2]
```diff
@@ -26,6 +26,47 @@ def __init__(self, line, character, offset=0):
...
+        return isinstance(__value, Position) and (self.line, self.character) < (
+            __value.line,
+            __value.character,
+        )
+
+    def __le__(self, __value: object) -> bool:
+        return isinstance(__value, Position) and (self.line, self.character) <= (
+            __value.line,
+            __value.character,
+        )
```
Comment (Nfsaavedra) [inline pylspclient/lsp_structs.py:68]: Here some of them could be simplified by defining them in relation to other operators (e.g. lt is just not ge)

File: README.md (hunk 1 above)
Comment (pcarrott) [reply README.md:29]: CoqFile no longer has the ast attribute, it only has steps where each Step has its own AST

Comment (Nfsaavedra) [reply README.md:29]: Oh ye true!
